nav2_tasks/include/nav2_tasks/navigate_to_pose_task.hpp

```diff
@@ -39,22 +40,46 @@ class NavigateToPoseTaskServer : public TaskServer<NavigateToPoseCommand, Naviga
     goal_sub_ = node_->create_subscription<geometry_msgs::msg::PoseStamped>("move_base_simple/goal",
         std::bind(&NavigateToPoseTaskServer::onGoalPoseReceived, this, std::placeholders::_1));
 
+    initial_pose_received_ = false;
+    initial_pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+      "initialpose",
+      std::bind(&NavigateToPoseTaskServer::onInitialPoseReceived, this, std::placeholders::_1));
+
     // A client that we'll use to send a command message to our own task server
     self_client_ = std::make_unique<nav2_tasks::NavigateToPoseTaskClient>(node_);
   }
 
   NavigateToPoseTaskServer() = delete;
 
+  bool hasInitialPoseReceived()
+  {
+    if (initial_pose_received_) {
+      return true;
+    }
+    return false;
+  }
+
+  void setInitialPose(bool initial_pose)
+  {
+    initial_pose_received_ = initial_pose;
+  }
+
 protected:
   // For backwards compatibility, the NavigateToPoseTaskServer will respond to the goal_pose
   // message sent from rviz. We'll receive the incoming message and invoke our own
   // NavigateToPose task using self_client_
   rclcpp::Subscription<geometry_msgs::msg::PoseStamped>::SharedPtr goal_sub_;
+  rclcpp::Subscription<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr initial_pose_sub_;
   std::unique_ptr<nav2_tasks::NavigateToPoseTaskClient> self_client_;
+  bool initial_pose_received_;
   void onGoalPoseReceived(const geometry_msgs::msg::PoseStamped::SharedPtr pose)
   {
     self_client_->sendCommand(pose);
   }
+  void onInitialPoseReceived(const geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr /*msg*/)
+  {
```

Comment:
Why not store the message with the initial pose we've received?

Reply:
The initial pose msg is not needed here.